### PR TITLE
initial implementation with ext-mongodb

### PR DESCRIPTION
**Pull request by robinvdvleuten**
I've copied over the original class from the initial project. Just to take a peek at for now. There are still some small things to do before it can be merged;

- [x]  copy class from initial project
- [ ] remove reference to `App\MongoDB\Client`
- [x] test all the things

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    ],
+    "homepage" : "https://github.com/broadway/event-store-mongodb",
+    "autoload": {
+        "psr-4": {
+            "Broadway\\EventStore\\": "src"
+        }
+    },
+    "extra": {
+        "branch-alias": {
+            "dev-master": "1.0.x-dev"
```
Comment (othillo) [inline composer.json]: maybe start with 0.2.x-dev first?

File: src/MongoDBEventStore.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    public function load($id)
+    {
+        $cursor = $this->collection->find(['uuid' => (string) $id]);
+
+        $messages = [];
+        foreach ($cursor as $message) {
+            $messages[] = $this->denormalizeDomainMessage($message);
+        }
+
+        if (empty($messages)) {
```
Comment (othillo) [inline src/MongoDBEventStore.php]: can you move this to line 58 using if 0 === $cursor->count() ?

**Review by othillo — CHANGES_REQUESTED** [review]

**Review by othillo — APPROVED** [review]

**Comment by wjzijderveld** [comment]
Thanks @robinvdvleuten @wgorczyca and @othillo !

**Comment by robinvdvleuten** [comment]
Thanks for all the effort the last days guys! Too bad I haven't found the time to finish the PR.

**Pull request merged**